doc/nrf/libraries/bin/lwm2m_carrier/app_integration.rst

```diff
@@ -170,7 +175,7 @@ The server settings can put the LwM2M carrier library either in the normal mode
 
 * :kconfig:option:`CONFIG_LWM2M_SERVER_BINDING_CHOICE`:
 
-  * The binding can be either ``U`` (UDP) or ``N`` (non-IP).
+  * The binding can be either :c:macro:`LWM2M_CARRIER_SERVER_BINDING_UDP` or :c:macro:`LWM2M_CARRIER_SERVER_BINDING_NONIP`.
   * Leaving this configuration empty selects the default binding (UDP).
   * If non-IP binding is configured, :kconfig:option:`CONFIG_LWM2M_CARRIER_CUSTOM_URI` is not used and must be left empty.
```

Comment:
It is more the other way arround, if UDP is selected you need an URI.
Maybe we can also mention that in case of Non-IP you need a Non-IP APN, you can set a custom one using :kconfig:option:`CONFIG_LWM2M_CARRIER_CUSTOM_APN` (or the default APN needs to be Non-IP). You need an IP APN for UDP.
Technically you can set both flags but this is not something we want to test and promote. In this case it depends on the APN what we do.